### DISCUSSION TITLE
[COMMANDS] Deprecated in PHP 8.0: Required parameter $... follows optional parameter $name.

### DISCUSSION THREAD

**Discussion by sgc-fireball** [symfony/symfony#45869] [Q&A]
Hi,
in Symfony the Commands construct accept an optional $name parameter. But this is very ugly because php 8.0 deprecated the optional parameter before required paramenter (and it is good idea to do this!).

https://github.com/symfony/console/blob/6.1/Command/Command.php#L113

Now we can't not inject needed services via services.xml without changing the construct order!
Deprecated in PHP 8.0: Required parameter $.... follows optional parameter $name.

What to you thing about it?

**Comment by sgc-fireball** [comment]
```
<service id="AddressImportAddressesCommand" .....>
            <argument>we need to set a name, would be override later!</argument>
            <argument type="service" id="Doctrine\DBAL\Connection"/>
             .........
            <tag name="console.command"/>
        </service>
```
If we must use "set...." the class doesn't contains all informations to run in the consturct and changing the parameter order it is ultra ugly too.

**Comment by stof** [comment]
The Command constructor accepts *only* the optional name. So it is not before required parameters.

And when creating a constructor in a child class, nothing forces you to keep the optional name as first argument.
